source/PlayerInfo.cpp

```diff
@@ -988,10 +988,20 @@ Ship *PlayerInfo::Flagship()
 // Determine which ship is the flagship and return the shared pointer to it.
 const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 {
+	bool clearance = false;
+	if(planet)
+	{
+		for(const Mission &mission : missions)
+			if(mission.HasClearance(planet))
+			{
+				clearance = true;
+				break;
+			}
+	}
```

Comment:
This should probably be done inside the following if statement, since it's not necessary if `flagship` isn't a null pointer.

Reply:
Good call. Moving that.